contract/views/contract.xml

```diff
@@ -96,8 +96,75 @@
                     <notebook>
                         <page name="recurring_invoice_line"
                               string="Recurring Invoices">
-                            <field name="contract_line_ids" attrs="{'readonly': [('is_terminated','=',True)]}"
-                                   context="{'default_contract_type': contract_type}"/>
+                            <field name="contract_line_ids"
+                                   attrs="{'readonly': [('is_terminated','=',True)]}"
+                                   widget="section_and_note_one2many"
+                                   context="{'default_contract_type': contract_type}">
+                                <tree decoration-muted="is_canceled"
+                                      decoration-info="create_invoice_visibility and not is_canceled">
+                                    <control>
+                                        <create string="Add a line"/>
+                                        <create string="Add a section" context="{'default_display_type': 'line_section'}"/>
+                                        <create string="Add a note" context="{'default_display_type': 'line_note'}"/>
+                                    </control>
+                                    <field name="display_type" invisible="1"/>
```

Comment:
Why do you need to declare inline tree view? The idea is to keep the isolated one.

Reply:
These <control> tags can only be defined in embedded tree views. If I defined that in an isolated tree view we get an error